### PR TITLE
Adjust ingest/extract scatter calculations for exomes [VS-1146]

### PR DESCRIPTION
**Pull request by rsasch**
[10 sample run](https://app.terra.bio/#workspaces/gvs-dev/GVS_190k_Exomes/job_history/10d083c7-2c20-4339-aa2b-70945056de44)
[5k sample run](https://app.terra.bio/#workspaces/gvs-dev/GVS_190k_Exomes/job_history/60325b52-7040-492b-82e1-23a58850fb59)
[20k sample run](https://app.terra.bio/#workspaces/gvs-dev/GVS_190k_Exomes/job_history/7fee1590-d00c-4038-a575-fd06a9edba1a)
[50k sample run](https://app.terra.bio/#workspaces/gvs-dev/GVS_190k_Exomes/job_history/2efb51a2-105a-4f62-a1fe-26b7350706ed)
[100k sample run](https://app.terra.bio/#workspaces/gvs-dev/GVS_190k_Exomes/job_history/dba1e793-51e4-47cc-9016-c478628e7252)
[integration run](https://app.terra.bio/#workspaces/gvs-dev/GVS%20Integration/job_history/19c865ab-69bc-45a3-a897-ff4e5f3d2a53)

### CODE REVIEW

File: scripts/variantstore/wdl/GvsExtractCallset.wdl [hunk 1]
```diff
@@ -110,12 +111,11 @@ workflow GvsExtractCallset {
...
-                                          else if GetNumSamplesLoaded.num_samples < 5000 then 1000
-                                               else if GetNumSamplesLoaded.num_samples < 20000 then 2000 # Stroke Anderson
-                                                    else if GetNumSamplesLoaded.num_samples < 50000 then 10000
-                                                         else if GetNumSamplesLoaded.num_samples < 100000 then 20000 # Charlie
-                                                              else 40000
+                                   else if GetNumSamplesLoaded.num_samples < 1000 then 500
+                                      else if GetNumSamplesLoaded.num_samples < 5000 then 1000
+                                         else if GetNumSamplesLoaded.num_samples < 20000 then 2000 # Stroke Anderson
+                                            else if GetNumSamplesLoaded.num_samples < 50000 then 10000
+                                              else if is_wgs then 20000 else 7500
```
Comment (koncheto-broad) [inline scripts/variantstore/wdl/GvsExtractCallset.wdl:118]: if I a reading this right, it looks like the only areas in which genomes and exomes need to differ in extract is above 50k samples? It also looks like the highest tier of scattering has been removed and scattering is capped at 20k now for wgs and 7500 for exomes.  Just want to clarify the desired behavior here.  Is that due to some of the scattering issues Kevin is seeing while he's attempting this pathway for pgen? Our behavior in this range seemed to be a bit off

Comment (mcovarr) [reply scripts/variantstore/wdl/GvsExtractCallset.wdl:118]: This is a bit surprising... if we're doing between 20K and 50K exomes we'd scatter 10K wide, but above 50K exomes and we'd only scatter 7500?

Comment (mcovarr) [reply scripts/variantstore/wdl/GvsExtractCallset.wdl:118]: Thank you for fixing the [bigger-than-Terra-would-allow limit of 40K](https://broadinstitute.slack.com/archives/C0544AAC70D/p1701205207833039) 🙏

Comment (rsasch) [reply scripts/variantstore/wdl/GvsExtractCallset.wdl:118]: The logic comes from past runs that failed/succeeded.  Basically, if the callsets are small enough, we don't need to scatter differently in VCF extract for WGS vs. exomes (that's not to say it's optimized; that was not part of this ticket).  We also don't support VCF extract above 98k whole genomes, so beyond a certain size, we only care about exomes getting extracted successfully.

Comment (koncheto-broad) [reply scripts/variantstore/wdl/GvsExtractCallset.wdl:118]: the checks out. I think we should likely rationalize it a bit in line with MIguel's observation, where we seem to actually *decrease* our scattering width at higher numbers.  We should likely either make 50k also 7500, up the top one to 10k, or maybe swap those two values

Comment (rsasch) [reply scripts/variantstore/wdl/GvsExtractCallset.wdl:118]: Not sure I understand the changes either of you are requesting.  We have created a 98K WGS callset, with extract scattered 20k and it worked.  We have created a 100k exome callset, with extract scattered 10k and 7500 and it worked.  What new callsets (how many samples and of what type) should I create and extract at what scatter?

Comment (koncheto-broad) [reply scripts/variantstore/wdl/GvsExtractCallset.wdl:118]: `                                            else if GetNumSamplesLoaded.num_samples < 50000 then 10000
                                              else if is_wgs then 20000 else 7500`

Just noting the contradictory nature of these two lines.  It appears as though by going ABOVE 50k samples we have to scatter to a LESSER amount.  Given the patterns of our scattering code elsewhere, this seems... odd.  And inconsistent.  Unless there are things going on that allow a 48k exome callset to scatter to 10k yet restrict a 75k exome callset to 7500?  It's purely the fact that the scatter width _dips_ for exomes in the highest bracket that is generating the questions.

If you feel as though this is accurate behavior, just make note of it in a comment (in the code) and move on.  If your statement about exomes working fine at 100k for both 10k and 7500, it sounds like 10k was probably quite generous for the 20k-50k band, and leaving it at 10k sounds probably reasonable too.  We can go tighten up numbers for the WES vs WGS paths more in a future ticket dedicated to optimizing for both paths.  At the very least though, just comment the code so somebody reading it doesn't look at it like we did and say "Why does the scatter width _reduce_ for exomes at the highest settings?" with no obvious answer

Comment (rsasch) [reply scripts/variantstore/wdl/GvsExtractCallset.wdl:119]: added a comment above the scatter logic

**Review by mcovarr — APPROVED** [review]

**Pull request merged**